docs: document dependencies in MAINTAIN.md (#15801)
neovim/neovim: MAINTAIN.md

    - The [nightly job](https://github.com/neovim/bot-ci/blob/master/ci/nightly.sh)
      will update the release assets based on the `stable` tag.

Third-party dependencies
--------------

The following are the third-party dependencies used in neovim.
  - [Klib](https://github.com/attractivechaos/klib)
  - [Lua](https://www.lua.org/download.html)
  - [LuaJIT](https://github.com/LuaJIT/LuaJIT)
  - [Luv](https://github.com/luvit/luv)
  - [libmpack](https://github.com/libmpack/libmpack)
  - [libtermkey](https://github.com/neovim/libtermkey)
  - [libuv](https://github.com/libuv/libuv)
  - [libvterm](http://www.leonerd.org.uk/code/libvterm/)
  - [lua-compat](https://github.com/keplerproject/lua-compat-5.3)
  - [tree-sitter](https://github.com/tree-sitter/tree-sitter)
  - [xdiff](https://github.com/git/git/tree/master/xdiff)

>   - [lua-cjson](https://github.com/openresty/lua-cjson)

Reviewer: there are 2 subcategories. 

1. The ones we need to manually update: lua-cjson, xdiff, mpack
2. The ones we need to "bump" in `third-party/CMakeLists.txt`

Reviewer: almost, klib is inlined so I would list it here also. Though it doesn't live in src/, we kinda forked it.

    - The [nightly job](https://github.com/neovim/bot-ci/blob/master/ci/nightly.sh)
      will update the release assets based on the `stable` tag.

Third-party dependencies
--------------

These "bundled" dependencies can be updated by bumping their versions in `third-party/CMakeLists.txt`:
  - [Lua](https://www.lua.org/download.html)
  - [LuaJIT](https://github.com/LuaJIT/LuaJIT)
  - [Luv](https://github.com/luvit/luv)
  - [libtermkey](https://github.com/neovim/libtermkey)
  - [libuv](https://github.com/libuv/libuv)
  - [libvterm](http://www.leonerd.org.uk/code/libvterm/)
  - [lua-compat](https://github.com/keplerproject/lua-compat-5.3)
  - [tree-sitter](https://github.com/tree-sitter/tree-sitter)

These dependencies are "vendored" (inlined), we need to update the sources manually:
  - [libmpack](https://github.com/libmpack/libmpack)
  - [xdiff](https://github.com/git/git/tree/master/xdiff)
  - [lua-cjson](https://github.com/openresty/lua-cjson)
  - [Klib](https://github.com/attractivechaos/klib)

> 

Reviewer: ```suggestion

We also maintain some forks, particularly for Windows, if we are waiting on upstream changes:
https://github.com/neovim/neovim/wiki/Deps

```

    - The [nightly job](https://github.com/neovim/bot-ci/blob/master/ci/nightly.sh)
      will update the release assets based on the `stable` tag.

Third-party dependencies
--------------

The following are the third-party dependencies used in neovim.

Dependencies that are managed in third-party/CMakeLists.txt:
  - [Lua](https://www.lua.org/download.html)
  - [LuaJIT](https://github.com/LuaJIT/LuaJIT)
  - [Luv](https://github.com/luvit/luv)
  - [libtermkey](https://github.com/neovim/libtermkey)
  - [libuv](https://github.com/libuv/libuv)
  - [libvterm](http://www.leonerd.org.uk/code/libvterm/)
  - [lua-compat](https://github.com/keplerproject/lua-compat-5.3)
  - [tree-sitter](https://github.com/tree-sitter/tree-sitter)

Dependencies which must be bumped manually:
  - [libmpack](https://github.com/libmpack/libmpack)
  - [xdiff](https://github.com/git/git/tree/master/xdiff)

>   - [lua-cjson](https://github.com/openresty/lua-cjson)

Reviewer: ```suggestion
  - [lua-cjson](https://github.com/openresty/lua-cjson)
  - [Klib](https://github.com/attractivechaos/klib)
```

    - The [nightly job](https://github.com/neovim/bot-ci/blob/master/ci/nightly.sh)
      will update the release assets based on the `stable` tag.

Third-party dependencies
--------------

> The following are the third-party dependencies used in neovim.
> 
> Dependencies that are managed in third-party/CMakeLists.txt:

Reviewer: ```suggestion
These "bundled" dependencies can be updated by bumping their versions in `third-party/CMakeLists.txt`:
```

>   - [Klib](https://github.com/attractivechaos/klib)

Reviewer: ```suggestion
```

  - [Lua](https://www.lua.org/download.html)
  - [LuaJIT](https://github.com/LuaJIT/LuaJIT)
  - [Luv](https://github.com/luvit/luv)
  - [libtermkey](https://github.com/neovim/libtermkey)
  - [libuv](https://github.com/libuv/libuv)
  - [libvterm](http://www.leonerd.org.uk/code/libvterm/)
  - [lua-compat](https://github.com/keplerproject/lua-compat-5.3)
  - [tree-sitter](https://github.com/tree-sitter/tree-sitter)

> Dependencies which must be bumped manually:

Reviewer: ```suggestion
These dependencies are "vendored" (inlined), we need to update the sources manually:
```
